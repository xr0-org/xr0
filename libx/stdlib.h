#ifndef STDLIB_H
#define STDLIB_H

axiom void *
malloc(int size) [ .alloc result; ];

axiom void
free(void *ptr) [ .dealloc ptr; ];

#endif
