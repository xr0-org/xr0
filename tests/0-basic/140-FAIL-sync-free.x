#include <stdlib.h>

void
unit(void *p)
{
	free(p);
}
