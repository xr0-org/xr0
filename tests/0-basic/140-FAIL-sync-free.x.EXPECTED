0-basic/140-FAIL-sync-free.x:6:9: cannot exec statement: dealloc on non-location
